src/Stratis.Bitcoin/P2P/PeerAddressManager.cs

```diff
@@ -157,12 +161,22 @@ public void AddPeers(NetworkAddress[] networkAddresses, IPAddress source)
         }
 
         /// <inheritdoc/>
-        public void PeerAttempted(IPEndPoint endpoint, DateTimeOffset peerAttemptedAt)
+        public void PeerAttempted(IPEndPoint endpoint, DateTime peerAttemptedAt)
         {
             var peer = this.FindPeer(endpoint);
             if (peer == null)
                 return;
 
+            //Reset the attempted count if:
+            //1: The last attempt was more than the threshold time ago.
+            //2: More than the threshold attempts was made.
+            if (peer.Attempted &&
```

Comment:
Do we need to check the `peer.Attempted` condition? I guess there is no harm.

Reply:
All the other set methods i.e. SetConnected / SetHandshaked are internal. I want to force the consumers outside of this assembly to go via the PeerAddressManager to set these values.